R/error_checking.R

```diff
@@ -0,0 +1,340 @@
+    if(is.null(raster_data))
+    {
+        return("Error: Raster data is NULL")
+    }
+    else if("RasterLayer" %in% class(raster_data))
+    {
+        raster_obj <- raster_data
+    }
+    else if("character" %in% class(raster_data))
+    {
+        if (!file.exists(raster_data))
+        {
+            return("Error: Cannon process raster file: Raster file does not exist or raster path is malformed.")
+        }
+    }
+    else
+    {
+        return("Error: Unrecognized raster_data argument.")
+    }
+
+    # Check raster_col argument to make sure it exists
+    if(is.null(raster_col)) # || raster_obj[raster_col] !exists
+    {
+        return("Error: No raster column defined")
+    }
+
+    # Verify raster CRS and assign default if NA
+    if(is.na(crs(raster_obj)) || is.null(crs(raster_obj)))
+    {
+        raster::crs(raster_obj) <- crs(default_projection)
+        print("Applying default CRS to raster (raster CRS was NA or NULL)")
+    }
+
+    return(result)
+}
+
+
+#' Verify map data
+#'
+#' This function runs a check on map data inputs and looks for errors
+#' @param map_data (Data Frame or Character) - A data frame that contains the output data to map, or alternatively a full path to a CSV
+#' @param data_key_field (Character) - Name of key field in data_obj for merging with shape_data
+#' @param data_col (Character) - Column name that contains the data object's output variable
+#' @return (Character) - Returns a success token or an error string if failed
+#' @export
+verify_data <- function(map_data, data_key_field, data_col)
+{
+    # Future
+    # 	Does it have nulls?
+    #  	Warning that pops up to the user about null values
+
+    result <- "Success"
+
+    # Verify map_data
+    if(!is.null(map_data))
+    {
+        # Verify data_key_field
+        if(!is.null(data_key_field))
+        {
+            if(!"character" %in% class(data_key_field))
+                return("Error: data_key_field argument must be of type character")
+            if(!data_key_field %in% colnames(map_data))
+                return("Error: data_key_field was not found in the map_data data frame")
+        }
+
+        # Verify data_col
+        if(!is.null(data_col))
+        {
+            if(!"character" %in% class(data_col))
+                return("Error: data_col argument must be of type character")
+            if(!data_col %in% colnames(map_data))
+                return("Error: data_col was not found in the map_data data frame")
+        }
+    }
+
+    return(result)
+}
+
+
+#' Verify csv data
+#'
+#' This function runs a check on csv data inputs and looks for errors
+#' @param map_data (Data Frame or Character) - Either the full path string to data file (csv) or a data.frame object
+#' @return (Character) - Returns a success token or an error string if failed
+#' @export
+verify_csv <- function(map_data)
+{
+
+    # Check csv for things like character Unicode
```

Comment:
don't export this unless it is ready to be used

Reply:
Add in TODO of docstring

Reply:
Fixed in previous commit- what do you mean by TODO?